CompositionRoot/B.swift

```diff
@@ -1,17 +1,17 @@
-protocol BPrototol {
+protocol BProtocol {
     func bMethod()
 }
 
-class B: BPrototol {
+class B: BProtocol {
     
-    private let applicationContext: ApplicationContext
+    private let cFactory: CFactoryProtocol
     
-    init(applicationContext: ApplicationContext) {
-        self.applicationContext = applicationContext
+    init(cFactory: CFactoryProtocol) {
```

Comment:
1. Remove the unnecessary injection of `ApplicationContext`.
2. Inject CFactory into B. B now depends on interfaces completely